WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -227,6 +227,8 @@ def post_results
     end
     post = Post.new(title: title, body: body, author: current_user, world_readable: true)
     post_post(post)
+    comp.results_posted = Time.now
+    comp.save!
```

Comment:
> Since `post_post()` returns the redirect or render, I think it needs to be last.

The order doesn't matter. `render` or `redirect` only alter a response object of that controller instance instead of returning a response. `render`'s return type is practically `void`.